framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/DatabaseDeploymentImpl.java

```diff
@@ -90,6 +94,17 @@ public String getNamespace() {
     }
 
     public String getServiceName() {
+        if (databaseName == null) {
+            // Database name not set, try to guess it from all available services
+            Pattern pattern = Pattern.compile(DATABASE_REGEXP);
```

Comment:
I'd initialize the regex directly in the static field:
`private static final Pattern DATABASE_REGEXP = Pattern.compile(...)`
but leave as it is if you prefer this way.